src/frontend/src/instance.rs

```diff
@@ -154,6 +155,9 @@ impl Instance {
             query_engine.clone(),
             dist_instance.clone(),
         ));
+        if let Some(configurator) = plugins.get::<Arc<Mutex<Option<Arc<StatementExecutor>>>>>() {
```

Comment:
I think we need to improve this by allowing `plugins.set::<Arc<StatementExecutor>>()` here. This can be done by changing plugins's `Arc` to `ArcSwap`.

Reply:
Also note that here we set `statement_executor` to plugins, it's not `configurator`